go/vt/topo/tablet.go

```diff
@@ -263,13 +267,21 @@ func (ts *Server) GetTabletsByCell(ctx context.Context, cellAlias string, opt *G
 		return nil, err
 	}
 
-	tablets := make([]*TabletInfo, len(listResults))
+	tablets := make([]*TabletInfo, 0)
 	for n := range listResults {
 		tablet := &topodatapb.Tablet{}
 		if err := tablet.UnmarshalVT(listResults[n].Value); err != nil {
 			return nil, err
 		}
-		tablets[n] = &TabletInfo{Tablet: tablet, version: listResults[n].Version}
+		if opt != nil && opt.Keyspace != "" {
+			if opt.Keyspace != tablet.Keyspace {
+				continue
+			}
+			if opt.Shard != "" && opt.Shard != tablet.Shard {
```

Comment:
This means Shard is ignored when Keyspace is not passed. That makes sense, I think. But that also means that I think we can improve the structure. There's already a struct/type that we can use:
```
type KeyspaceShard struct {
	Keyspace string
	Shard    string
}
```

We can add a member of that type to `GetTabletsByCellOptions` instead. Then we can also simplify the hint note above to:
```
var sizeHint int
if opt.KeyspaceShard == nil {
    sizeHint = len(listResults)
}
tablets := make([]*TabletInfo, 0, sizeHint)
```

Reply:
Nice! Updated